src/services/jsDoc.ts

```diff
@@ -92,9 +92,16 @@ namespace ts.JsDoc {
         // from Array<T> - Array<string> and Array<number>
         const parts: SymbolDisplayPart[][] = [];
         forEachUnique(declarations, declaration => {
-            for (const { comment } of getCommentHavingNodes(declaration)) {
-                if (comment === undefined) continue;
-                const newparts = getDisplayPartsFromComment(comment, checker);
+            for (const jsdoc of getCommentHavingNodes(declaration)) {
+                if (jsdoc.comment === undefined
+                    || isJSDoc(jsdoc)
+                       && declaration.kind !== SyntaxKind.JSDocTypedefTag

```

Comment:
Can you please explain what this clause does?

Reply:
`@typedef` tags are themselves declarations, and we want to show their comments when you explicitly request it. We only want to skip typedef comments for *other* declarations.

From point 1 in quickInfoJsDocTagsTypedef.ts:

```js
/** 
 * Bar comment
 * @typedef {Object} /*1*/Bar
 */
```

Edit: Definitely worth an explanation since both you and @andrewbranch had questions about this predicate.

Reply:
Your explanation makes sense and I agree that a comment would be helpful.

Reply:
Also, this additional check isn't needed in getJSDocTags because `@typedef` can't have child tags that are `@typedef`, only `@property`.